src/app/handlers/new_member_role.handler.ts

```diff
@@ -45,14 +25,11 @@ export class NewMemberRoleHandler implements IHandler {
   }
 
   private _pingUserInVerify(member: GuildMember) {
-    if (!this._VERIFY_CHANNEL) {
-      this._VERIFY_CHANNEL = this.container.guildService
-        .get()
-        .channels.filter((chan: GuildChannel) => chan.name === Constants.Channels.Bot.Verify)
-        .first() as TextChannel;
-    }
+    const verifyChannel = this.container.guildService.getChannel(
+      Constants.Channels.Bot.Verify
+    ) as TextChannel;
 
-    this._VERIFY_CHANNEL.send(member.user.toString()).then((sentMsg) => {
+    verifyChannel.send(member.user.toString()).then((sentMsg) => {
```

Comment:
await

Reply:
you could keep `_pingUserInVerify` non-async and then return the promise from `verifyChannel.send(...)` and then await that where `_pingUser...` is invoked